docs/lab/01.md

```diff
@@ -2,4 +2,376 @@
+
+Examples of actuators:
+- Fans    
+- Audible indicators (buzzers)
+- Light indicators
+- Heating resistors
+
+:::tip
+Sometimes, to activate an actuator, an actuating element is needed. For instance, to start a motor, the microcontroller simply sends a logical start command to a transistor that opens and allows a high current to pass through it (here, by "high current", we compare it to the maximum of a few milliamperes that a microcontroller can output).
+:::
+
+Examples of sensors:
+
+- Buttons
+- Photo resistors - their electrical resistance is influenced by the amount of light
+- Thermistors - their electrical resistance is influenced by temperature
+
+:::tip
+Depending on the type of transducers, they may require signal processing before being taken in by the microcontroller (signal conditioning). For example, a photo resistor needs to be used in a circuit with a voltage divider or a current source. Alternatively, some sensors can be connected directly to the microcontroller, such as buttons.
+:::
+
+### LEDs
+
+LEDs - Light Emitting Diode - also called electroluminescent diodes - emit light when they are directly polarized. Not to be confused with light bulbs as they have radically different methods of operation.   
+
+
+LEDs can be used as indicator lights (often used in various appliances to signal that the appliance is on and doing something), or for illumination, in which case power LEDs are used. In the lab, LEDs are used to indicate the status of a pin.
+
+#### Calculation of current limiting resistor
+
+
+To use an LED for the purpose of indicating the status of a pin (rather said to indicate the presence of voltage), the current through the LED must be limited. This can be done most simply by stringing a resistor with the LED.   
+
+An LED is designed to operate at a nominal current (ex: 10mA). The voltage drop at this current across low power indicator LEDs is given by the chemistry of the LED (this also gives the color of the LED). In the lab, since we are using such a low current LED, we can power it directly from the logic pins of the MCU.
+
+<div align="center">
+![LED](images/led.png)
+</div>
+
+$$
+R_{1} = \frac{(U_{pin} - U_{led})}{I_{led}}
+$$
+
+#### Example:
+
+If the MCU has a pin voltage of 3.3V, also noted as 3V3, to light up an LED with a nominal current of 10mA and a voltage drop of 2V we need 
+a resistance of 130 &ohm;.
+
+:::note
+We can use a resistor with a higher resistance value. The nominal current will light up the LED at it's maximum brightness. For status LEDs we can pick a resistance even 10 times bigger and the LED will light up slightly. 
+:::
+
+:::danger
+If there is no resistor in the circuit, the resistance will be almost `0`, the current will tend to `infinity`, meaning a short circuit. This will absolutely burn the LED and make it unusable, but it can also burn the MCU. Most MCUs have short circuit protection, but is safer to not rely on that.  
```

Comment:
```suggestion
If there is no resistor in the circuit, the resistance will be almost 0 &ohm;, the current will tend to `infinity`, meaning a short circuit. This will absolutely burn the LED and make it unusable, but it can also burn the MCU. Most MCUs have short circuit protection, but is safer to not rely on that.  
```